File: RemoteMonitor Desktop/src/main/java/components/GPU.java
package components;

import com.profesorfalken.jsensors.JSensors;
import com.profesorfalken.jsensors.model.components.Components;
import com.profesorfalken.jsensors.model.components.Gpu;
import com.profesorfalken.jsensors.model.sensors.Load;
import oshi.hardware.GraphicsCard;

import java.util.List;

public class GPU extends Component{
    private long vramTotal;
    private transient GraphicsCard graphicsCard;
    private transient Gpu jgpu;
    public GPU(GraphicsCard gpu, Gpu gpu2)
    {
        graphicsCard = gpu;
        vramTotal = graphicsCard.getVRam();
        componentName = graphicsCard.getName();
        componentType = "GPU";
        jgpu = gpu2;

    }
    public void update()
    {
        try {
            temperature = jgpu.sensors.temperatures.get(0).value;
        }
        catch (Exception e){}
        usage = jgpu.sensors.loads.get(0).value;
        lastUpdated = System.currentTimeMillis();
    }

    public long getVramTotal() {
        return vramTotal;
    }
}

File: RemoteMonitor Mobile/RemoteMonitor/RemoteMonitor/RemoteMonitor/Components/GPU.cs
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Text;

namespace RemoteMonitor.Components
{
    public class GPU : Component
    {
        public long vramTotal;
        public GPU(string name, double temp, double componentUsage, int vRam)
        {
            componentName = name;
            componentType = "GPU";
            update(temp, componentUsage, vRam);
        }
        public GPU() 
        {
            componentName = "No GPU detected";
            temperature = 0;
            usage = 0;
            vramTotal = 0;
        }
        public GPU(JObject json)
        {
            componentName = json["componentName"].ToString();
            componentType = "GPU";
            temperature = json["temperature"].ToObject<double>();
            usage = json["usage"].ToObject<double>();
            vramTotal = json["vramTotal"].ToObject<long>();
            update(json);
        }
        public void update(JObject json)
        {
            temperature = json["temperature"].ToObject<double>();
            usage = json["usage"].ToObject<double>();
            vramTotal = json["vramTotal"].ToObject<long>();
        }
        public void update(double temp, double componentUsage, int vRam)
        {
            temperature = temp;
            usage = componentUsage;
            vramTotal = vRam;
        }
        public long getVramTotal()
        {
            return vramTotal;
        }

    }
}
